src/main/java/mops/foren/applicationservices/PostService.java

```diff
@@ -0,0 +1,35 @@
+package mops.foren.applicationservices;
+
+import mops.foren.domain.model.Post;
+import mops.foren.domain.model.User;
+import mops.foren.domain.repositoryabstraction.IPostRepository;
+
+import java.util.List;
+
+public class PostService {
+
+    private IPostRepository postRepository;
+
+    public PostService(IPostRepository postRepository) {
+        this.postRepository = postRepository;
+    }
+
+    public List<Post> getPosts(Long threadId) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Method to add a post.
+     *
+     * @param post     The post to add
+     * @param user     The user that wants to create the post
+     * @param threadId The threadId the post belongs to
+     */
+    public void addPost(Post post, User user, Long threadId) {
```

Comment:
Look into branch "permissions-and-new-domain-model". I wrapped the Id's so the Method checkPermission will know if the id is a ThreadId or ForumId.